### PR TITLE
Duet AI Prompts: Apache Beam Runners 

### PR DESCRIPTION
**Pull request by dariabezkorovaina**
- Adding 3 new prompts for runners: Spark, Dataflow, Direct. 
- Fixing minor typos and improvements to the previously merged prompts 

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

File: learning/prompts/documentation-lookup/45_spark_runner.md [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+<dependency>
+  <groupId>org.apache.spark</groupId>
+  <artifactId>spark-streaming_2.12</artifactId>
+  <version>${spark.version}</version>
+</dependency>
+```
+
+Additionally, shade the application JAR using the Maven shade plugin by executing the following code:
+
+```java
```
Comment (olehborysevych) [inline learning/prompts/documentation-lookup/45_spark_runner.md:57]: @dariabezkorovaina I would just provide a list of required steps and point to documentation. This looks like too m,uch information to grasp in a single response.

File: learning/prompts/documentation-lookup/46_direct_runner.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+```java
+java -jar your-project.jar
+```
+
+***Considerations:***
+
+1. The Direct Runner is not intended for production pipelines as it prioritizes correctness over performance.
+2. The Direct Runner must fit all user data in memory, unlike other runners that can spill data to disk. Run it with small datasets that can fit within available memory.
+3. For unbounded data sources or sinks, set the `streaming` option to `true`. Streaming support for Python Direct Runner is limited. For the list of known issues, refer to [the GitHub issue #24528](https://github.com/apache/beam/issues/24528) in the Apache Beam repository.
```
Comment (olehborysevych) [inline learning/prompts/documentation-lookup/46_direct_runner.md:71]: I don't think we need this here

File: learning/prompts/documentation-lookup/46_direct_runner.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+5. You can control the number of threads or subprocesses by setting the `targetParallelism` option for Java or the `direct_num_workers` option for Python.
+
+For testing Apache Beam pipelines with the Direct Runner, refer to:
+[Testing Unbounded Pipelines in Apache Beam](https://beam.apache.org/blog/2016/10/20/test-stream.html) discusses using Java classes [`PAssert`](https://beam.apache.org/releases/javadoc/current/index.html?org/apache/beam/sdk/testing/PAssert.html) and [`TestStream`](https://beam.apache.org/releases/javadoc/current/index.html?org/apache/beam/sdk/testing/TestStream.html).
+The [Apache Beam WordCount Walkthrough](https://beam.apache.org/get-started/wordcount-example/#testing-your-pipeline-with-asserts) contains an example of logging and testing a pipeline with [`PAssert`](https://beam.apache.org/releases/javadoc/current/index.html?org/apache/beam/sdk/testing/PAssert.html) (Java) and [`assert_that`](https://beam.apache.org/releases/pydoc/current/apache_beam.testing.util.html#apache_beam.testing.util.assert_that) (Python).
```
Comment (olehborysevych) [inline learning/prompts/documentation-lookup/46_direct_runner.md:77]: i don't think this is related

File: learning/prompts/documentation-lookup/45_spark_runner.md [hunk 4]
```diff
@@ -0,0 +1,188 @@
+Prompt: How to execute Apache Beam pipelines with Apache Spark?
```
Comment (olehborysevych) [inline learning/prompts/documentation-lookup/45_spark_runner.md:1]: I think we should try to make response more concise. if possible

**Review by olehborysevych — APPROVED** [review]
LGTM

**Comment by dariabezkorovaina** [comment]
@damccorm please review the new batch of Duet AI training prompts.

File: learning/prompts/documentation-lookup/44_dataflow_runner.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+You can utilize pipeline options to control various aspects of how Cloud Dataflow executes your job. For instance, you can specify whether your pipeline runs on worker virtual machines, on the Cloud Dataflow service backend, or locally. For additional pipeline configuration options, refer to the reference documentation for the [`PipelineOptions`](https://beam.apache.org/releases/pydoc/current/apache_beam.options.pipeline_options.html#apache_beam.options.pipeline_options.PipelineOptions) interface (Python) or the [`DataflowPipelineOptions`](https://beam.apache.org/releases/javadoc/current/index.html?org/apache/beam/runners/dataflow/options/DataflowPipelineOptions.html) interface (Java).
```
Comment (damccorm) [inline learning/prompts/documentation-lookup/44_dataflow_runner.md:39]: For Python, this should probably link to `GoogleCloudOptions` for consistency - https://beam.apache.org/releases/pydoc/current/apache_beam.options.pipeline_options.html#apache_beam.options.pipeline_options.GoogleCloudOptions

File: learning/prompts/documentation-lookup/44_dataflow_runner.md [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+
+***6. Block Execution or Cancel The Job:***
+
+To block execution, use the [`waitUntilFinish`](https://beam.apache.org/releases/javadoc/current/org/apache/beam/sdk/PipelineResult.html#waitUntilFinish–) method for Java or the [`wait_until_finish`](https://beam.apache.org/releases/pydoc/current/apache_beam.runners.runner.html#apache_beam.runners.runner.PipelineResult.wait_until_finish) method for Python. The methods wait until the pipeline finishes and return its final status. The Cloud Dataflow Runner prints job status updates and console messages while waiting for the pipeline to finish.
+
+To cancel the job, you can use the [Dataflow Monitoring Interface](https://cloud.google.com/dataflow/pipelines/dataflow-monitoring-intf) or the [Dataflow Command-line Interface](https://cloud.google.com/dataflow/pipelines/dataflow-command-line-intf).
+
+For more information on the supported capabilities of the Cloud Dataflow Runner, refer to the [Beam Capability Matrix](https://beam.apache.org/documentation/runners/capability-matrix/) in the Apache Beam documentation.
```
Comment (damccorm) [inline learning/prompts/documentation-lookup/44_dataflow_runner.md:85]: I think we should just cut this section, it doesn't seem core to answering the main question

**Review by damccorm — COMMENTED** [review]
Mostly looks good to me, just had a couple minor comments

**Comment by dariabezkorovaina** [comment]
@damccorm thanks very much:) Implemented the changes, please take a look.

**Review by damccorm — APPROVED** [review]
Thanks!

**Pull request merged**